apis/kueue/v1alpha1/multikueue_types.go

```diff
@@ -55,6 +59,14 @@ type KubeConfig struct {
 }
 
 type MultiKueueClusterSpec struct {
+	// controllerName is name of the controller which will actually perform
+	// the checks. This is the name with which controller identifies with,
+	// not necessarily a K8S Pod or Deployment name. Cannot be empty.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:default="kueue.x-k8s.io/multikueue"
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf", message="field is immutable"
+	ControllerName string `json:"controllerName"`
```

Comment:
We deliberately didn't make it configurable if the first iteration to limit the potential of misconfiguring the system. Could you first open the issue to discuss the need for it?

Reply:
The status of a multikueue cluster follows the state of a contreller-runtime client in the kueue-controller-manager, if we want to be able to have multiple controller managers working with the clusters we need a way to mark who is managing each cluster.

Reply:
I see, but I'm wondering if we could alternatively still use MultiKueue controller, but have a concept of external plugin? 

To make it easier to track of an object is managed by MultiKueue or not. I feel this would help OnCall, but maybe I need to think about this more.

Since this required API changes should we not first go via KEP to discuss alternatives? WDYT @alculquicondor, @tenzen-y?

Reply:
> if we want to be able to have multiple controller managers working with the clusters we need a way to mark who is managing each cluster.

There should only be one controller manager modifying these objects (Kueue).

Other controller managers should only be in charge of mirroring the Job CRDs, but otherwise they can use all the same objects.

I would like to see at least a description of what the plan is, how components should interact, what each of them is responsible of. Maybe a KEP is worth, but please start with some description before writing more code.

Reply:
> I see, but I'm wondering if we could alternatively still use MultiKueue controller, but have a concept of external plugin?

That will be a lot more complicated and will most likely require some additional API object type(s) to communicate  between that custom controller to multikueue admission check controller. 

> There should only be one controller manager modifying these objects (Kueue).

MultiKueueClusters should not be that different then AdmissionChecks, which by design are managed by an AdmissionCheckController running in or outside of Kueue.

Reply:
The current approach will lose the consistency between the kueue-manager and external controllers since it seems that the current approach allows us to re-implement the multikueue admission controller.

Can we consider an alternative approach to prevent the losing specification consistency?

As I described in the #2349, nice to start in the synchronizing design and the actual implementations.